### PR TITLE
JDBC Value mapping

### PR DESCRIPTION
**Pull request by VardhanThigle**
Implementing `JDBCIO.RowMapper` to map ResultSet to SourceRow.

### CODE REVIEW

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/JdbcSourceRowMapper.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+   * @param mappingsProvider Mapping Provider based on the type of database.
+   * @param sourceTableSchema Schema of source table.
+   */
+  public JdbcSourceRowMapper(
+      JdbcValueMappingsProvider mappingsProvider, SourceTableSchema sourceTableSchema) {
+    this.mappingsProvider = mappingsProvider;
+    this.sourceTableSchema = sourceTableSchema;
+  }
+
+  long getCurrentTime() {
```
Comment (Deep1998) [inline v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/JdbcSourceRowMapper.java:54]: nit: getCurrentTimeInMicros?

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/JdbcSourceRowMapper.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                  schema = schema.getTypes().get(1);
+                }
+                builder.setField(
+                    entry.getKey(),
+                    this.mappingsProvider
+                        .getMappings()
+                        .getOrDefault(entry.getValue().getName(), JdbcValueMapper.UNSUPPORTED)
+                        .mapValue(resultSet, entry.getKey(), schema));
+              } catch (SQLException e) {
+                logger.error(
```
Comment (Deep1998) [inline v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/JdbcSourceRowMapper.java:92]: The error msg could be more informative. We can include details like the specific field and the expected data type to aid debugging.

Comment (VardhanThigle) [reply v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/JdbcSourceRowMapper.java:92]: The resultSet exception will provide the necessary information.

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/JdbcValueMapper.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ */
+public class JdbcValueMapper<T extends Object> {
+
+  private ResultSetValueExtractor<T> valueExtractor;
+  private ResultSetValueMapper<T> valueMapper;
+
+  /**
+   * Construct the {@link JdbcValueMapper}.
+   *
+   * @param resultSetValueExtractor Extractor to extreact the desired field from resultType.
```
Comment (Deep1998) [inline v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/JdbcValueMapper.java:36]: extreact*

**Review by Deep1998 — APPROVED** [review]

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/provider/package-info.java [hunk 4]
```diff
@@ -0,0 +1 @@
+package com.google.cloud.teleport.v2.source.reader.io.jdbc.rowmapper.provider;
```
[the remark this answers is not in the document]
Comment (VardhanThigle) [reply v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/provider/package-info.java:20]: Thanks @damccorm . Added the same.

**Pull request merged**